Lib/subprocess.py

```diff
@@ -218,22 +218,39 @@ def __repr__(self):
         _PopenSelector = selectors.SelectSelector
 
 
-# This lists holds Popen instances for which the underlying process had not
-# exited at the time its __del__ method got called: those processes are wait()ed
-# for synchronously from _cleanup() when a new Popen object is created, to avoid
-# zombie processes.
-_active = []
-
-def _cleanup():
-    for inst in _active[:]:
-        res = inst._internal_poll(_deadstate=sys.maxsize)
-        if res is not None:
-            try:
-                _active.remove(inst)
-            except ValueError:
-                # This can happen if two threads create a new Popen instance.
-                # It's harmless that it was already removed, so ignore.
-                pass
+if _mswindows:
+    # On Windows we just need to close `Popen._handle` when we no longer need
+    # it, so that the kernel can free it. Keeping the handle open will actually
+    # create a zombie until the next `_cleanup()` call, which may be never if
+    # `Popen()` isn't called again. `CloseHandle` is called as a
+    # `Popen._handle.__del__`, so there is no need to collect it as `_active`
+    # in `Popen.__del__` if it didn't finish yet.
+    #
```

Comment:
I don't think it helps to talk about zombies here. It's not a visible zombie in the task list, unlike Unix. The process has already been rundown in the kernel. What remains, as long as it's kept alive by a handle or pointer reference, is the kernel Process object that has information such as the exit status. What we may need to clarify for the reader is that `Popen._handle` gets closed implicitly when the `Popen` instance is finalized, so there is nothing for `_cleanup` to do in Windows.